### PR TITLE
Add "All" getters to segment

### PR DESCRIPTION
**Pull request by jakejohns**
Add methods to Segment:
- getAll()
- getFlashAll()
- getFlashNextAll()

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -4,4 +4,9 @@
             <directory>./tests</directory>
         </testsuite>
     </testsuites>
+    <filter>
```
Comment (harikt) [inline phpunit.xml.dist:7]: Sorry I missed to see you have already added this. Was looking at travis and it notifies the problem and ie why ended up adding in #50 .

File: README.md [hunk 2]
```diff
@@ -114,6 +118,9 @@ Then, in subsequent requests, we can read the flash value using `getFlash()`:
 <?php
 $segment = $session->getSegment('Vendor\Package\ClassName');
 $message = $segment->getFlash('message'); // 'Hello world!'
+
+// You can also get all the flash messages
+$messages = $segment->getFlashAll();
```
Comment (harikt) [inline README.md:123]: I prefer to call `getAllFlash` . Don't think `getFlashAll` looks good :( .

File: src/SegmentInterface.php [hunk 3]
```diff
@@ -32,6 +32,17 @@ public function get($key, $alt = null);
 
     /**
      *
+     * Returns array of all key/value pairs in the segment.
+     *
+     * @param mixed $alt An alternative value to return if the segment name is not set.
+     *
```
Comment (harikt) [inline src/SegmentInterface.php:38]: Don't change interface. Things will break. In that case we may need to go with 3.x .

**Comment by jakejohns** [comment]
@harikt , Good point!
That was silly of me not to think that changing the Interface would be a breaking change. 
I'll go ahead and change the method names to read more natural, instead of group alphabetical.

It probably doesnt make sense to add to Segment without changing the Interface, right?
So I should also probably look at the 3.x branch too, right? Or what do you think?

**Comment by harikt** [comment]
Yes probably you could check 3.x branch may be that is good. Ping @pmjones for more information where it should go.

**Comment by jakejohns** [comment]
Closing in favor of #52 since the change in the interface is a breaking change.

**Pull request closed without merging**